client/src/main/java/org/spine3/client/QueryOperators.java

```diff
@@ -0,0 +1,187 @@
+ *     <li>other operators currently support only {@link Timestamp} instances comparison; if any of
+ *         the operands is {@code null}, the comparison returns {@code false}.
+ * </ol>
+ *
+ * <p>Support for some other data types may be added in future.
+ *
+ * @author Dmytro Dashenkov
+ */
+@Internal
+public final class QueryOperators {
+
+    private QueryOperators() {
+        // Prevent this static class from being initialized.
+    }
+
+    /**
+     * Compares the given operands with the given operator.
+     *
+     * <p>For example, if operands where {@code 42} and {@link 9} (exactly in that order) and
+     * the operator was {@link Operator#GREATER_THAN GREATER_THAN}, then this function could be
+     * expressed as {@code 42 > 8}. The function returns {@code true} if the expression is
+     * mathematically correct.
+     *
+     * @param left     the left operand
+     * @param operator the comparison operator
+     * @param right    the right operand
+     * @param <T>      the type of the compared values
+     * @return {@code true} if the operands match the operator, {@code false} otherwise
+     * @throws UnsupportedOperationException if the operation is
+     *                                       <a href="supported_types">not supported</a> for
+     *                                       the given data types
+     */
+    public static <T> boolean compare(@Nullable T left, Operator operator, @Nullable T right)
+            throws UnsupportedOperationException {
+        checkNotNull(operator);
+        final MetaOperator op = getMetaInstance(operator);
+        final boolean result = op.compare(left, right);
+        return result;
+    }
+
+    /**
+     * @return a symbolic representation of the given {@link Operator}
+     */
+    public static String toString(Operator operator) {
+        checkNotNull(operator);
+        final MetaOperator op = getMetaInstance(operator);
+        return op.toString();
+    }
+
+    private static MetaOperator getMetaInstance(Operator operator) {
+        final String name = operator.name();
+        try {
+            final MetaOperator metaOperator = MetaOperator.valueOf(name);
+            return metaOperator;
+        } catch (@SuppressWarnings("ProhibitedExceptionCaught") // Thrown by Enum on missing value
+                NullPointerException npe) {
+            throw new IllegalArgumentException(
+                    format("Cannot recognize query operator %s.", name),
+                    npe);
+        }
+    }
+
+    /**
+     * A reflected {@link Operator} enum facilitated by custom methods.
+     *
+     * <p>Since {@link Operator} is defined in Protobuf, we cannot add custom methods to this type
+     * in Java. The way abound is using this type, whose enum constants have the same name as
+     * the {@link Operator} constants do.
+     *
+     * @see QueryOperators#getMetaInstance(Operator)
+     */
+    private enum MetaOperator {
```

Comment:
I don't like an idea of duplicating the operators. We could use the proto-defined enum and fill our own operators in Java.

Let's discuss vocally.